register/views.py

```diff
@@ -1,8 +1,68 @@
-from registration.backends.default.views import ActivationView
+from django.views.generic import TemplateView, FormView, RedirectView
+from django.shortcuts import render
+from django.contrib.auth import get_user_model, login, logout
+from django.urls import reverse, reverse_lazy
+
 from registration.backends.default.views import RegistrationView
 
-from django.views.generic import TemplateView
+from . import forms
 
 
 class ProfileView(TemplateView):
     template_name = 'register/profile.html'
+
+
+class EmailView(FormView):
+    template_name = 'registration/email.html'
+    success_url_login = reverse_lazy('auth_login')
+    success_url_register = reverse_lazy('auth_register')
+    form_class = forms.EmailForm
+
+    def form_valid(self, form):
+        self.request.session['email'] = form.data.get('email')
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        email = self.request.session['email']
+        User = get_user_model()
+        try:
+            User.objects.get(email=email)
+        except User.DoesNotExist:
+            return self.success_url_register
+        return self.success_url_login
+
+
+class EmailPrefilledMixin:
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        email = self.request.session.get('email')
```

Comment:
Here is another use, that's why I'm not sure about pop-ing it but this mixin shouldn't interfere if I understand correctly?